lib/aptible/cli/helpers/s3_log_helpers.rb

```diff
@@ -0,0 +1,201 @@
+require 'aws-sdk-s3'
+require 'pathname'
+
+module Aptible
+  module CLI
+    module Helpers
+      module S3LogHelpers
+        def ensure_aws_creds
+          cred_errors = []
+          unless ENV['AWS_ACCESS_KEY_ID']
+            cred_errors << 'Missing environment variable: AWS_ACCESS_KEY_ID'
+          end
+          unless ENV['AWS_SECRET_ACCESS_KEY']
+            cred_errors << 'Missing environment variable: AWS_SECRET_ACCESS_KEY'
+          end
+          raise Thor::Error, cred_errors.join(' ') if cred_errors.any?
+        end
+
+        def validate_log_search_options(options = {})
+          id_options = [
+            options[:app_id],
+            options[:database_id],
+            options[:proxy_id],
+            options[:container_id]
+          ]
+          date_options = [options[:start_date], options[:end_date]]
+          unless options[:string_matches] || id_options.any?
+            m = 'You must specify an option to identify the logs to download,' \
+                ' either: --string-matches, --app-id, --database-id,' \
+                ' --proxy-id, or --container-id'
+            raise Thor::Error, m
+          end
+
+          m = 'You cannot pass --app-id, --database-id, --proxy-id, or ' \
+              '--container-id when using --string-matches.'
+          raise Thor::Error, m if options[:string_matches] && id_options.any?
+
+          m = 'You must specify only one of ' \
+              '--app-id, --database-id, --proxy-id or --container-id'
+          raise Thor::Error, m if id_options.any? && !id_options.one?
+
+          m = 'The options --start-date/--end-date cannot be used when ' \
+              'searching by string'
+          raise Thor::Error, m if options[:string_matches] && date_options.any?
+
+          m = 'You must pass both --start-date and --end-date'
+          raise Thor::Error, m if date_options.any? && !date_options.all?
+        end
+
+        def info_from_path(file)
+          properties = {}
```

Comment:
minor nit (and probably preference), but instead of repeatedly splitting the path for a given variable, can't we just reference the  the file path with destructuring (something like below)?

```rb
file_path_elements = file.split("/")
_, properties[:schema], properties[:shasum], type_id = file_path_elements
```

and then conditionally accessing 5/6 positions from that `file_path_elements` instead of repeatedly generating a bunch of arrays? (as these should all reference the first `file_path_elements`

Reply:
I do like your suggestion actually. I think your solution is the obvious code-first solution, whereas i was thinking more human-first about the file name.

Reply:
it's preference, but i leave it to you. sometimes this notation can be helpful and i've notice rubocop gets really angry when your functions get long which gives you a finish line to be super paranoid about at all times